Issue 1210 update pcd prl

roberto-ebizmarts opened the pull request:
closes #1210

roberto-ebizmarts committed:
Promo Rules are unmarked as modified from ecommerce items table once Mailchimp confirms they were changed.
Promo Rules are modified now instead of being deleted.

roberto-ebizmarts committed:
Promo Codes are marked as modified from ecommerce once they are modified.
Creating properly PATCH requests for PRL an PCD.

roberto-ebizmarts committed:
Promo Codes are being sent with "enabled" body param which matches with the "is_active" field of related Promo Rule.

roberto-ebizmarts committed:
Merge branch 'develop' of github.com:mailchimp/mc-magento into issue_1210_update_PCD_PRL

roberto-ebizmarts committed:
Promo Codes are being sent with "enabled" body param which matches with the "is_active" field of related Promo Rule.

gonzaloebiz commented on app/code/community/Ebizmarts/MailChimp/Model/Api/PromoCodes.php:
```diff
@@ -196,6 +197,75 @@ protected function _getNewPromoCodes()
+            Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE,
+            "m4m.mailchimp_sync_modified = 1",
+            "modified"
+        );
```
You must use the mailchimpStoreId in the collection

gonzaloebiz committed:
add mailchimp_store_id to the modified collection
unify the generateCodeData function
#1210

roberto-ebizmarts committed:
Using existent generateCodeData() function, instead: _generatePromoCodeData()
Removing unnecessary _makeModifiedPromoCodesCollection() function
Passing $magentoStoreId instead $mailchimpStoreId to generateCodeData() function
